test/integration/sequelize.transaction.test.ts

```diff
@@ -0,0 +1,167 @@
+
+  afterEach(() => {
+    for (const stub of stubs) {
+      stub.restore();
+    }
+
+    stubs = [];
+  });
+
+  describe('Transaction#commit', () => {
+    it('returns a promise that resolves once the transaction has been committed', async () => {
+      const t = await sequelize.transaction();
+
+      await expect(t.commit()).to.eventually.equal(undefined);
+    });
+
+    // we cannot close a sqlite connection, but there also cannot be a network error with sqlite.
+    // so this test is not necessary for that dialect.
+    if (dialectName !== 'sqlite') {
+      it('does not pollute the pool with broken connections if commit fails', async () => {
+        const initialPoolSize = sequelize.connectionManager.pool.size;
+
+        stubs.push(sinon.stub(sequelize.queryInterface, 'commitTransaction').rejects(new Error('Oh no, an error!')));
+
+        const t = await sequelize.transaction();
+
+        await expect(t.commit()).to.be.rejectedWith('Oh no, an error!');
+
+        // connection should have been destroyed
+        expect(sequelize.connectionManager.pool.size).to.eq(Math.max(0, initialPoolSize - 1));
+      });
+    }
+  });
+
+  describe('Transaction#rollback', () => {
+    it('returns a promise that resolves once the transaction has been rolled back', async () => {
+      const t = await sequelize.transaction();
+
+      await expect(t.rollback()).to.eventually.equal(undefined);
+    });
+
+    // we cannot close a sqlite connection, but there also cannot be a network error with sqlite.
+    // so this test is not necessary for that dialect.
+    if (dialectName !== 'sqlite') {
+      it('does not pollute the pool with broken connections if the rollback fails', async () => {
+        const initialPoolSize = sequelize.connectionManager.pool.size;
+
+        stubs.push(sinon.stub(sequelize.queryInterface, 'rollbackTransaction').rejects(new Error('Oh no, an error!')));
+
+        const t = await sequelize.transaction();
+
+        await expect(t.rollback()).to.be.rejectedWith('Oh no, an error!');
+
+        // connection should have been destroyed
+        expect(sequelize.connectionManager.pool.size).to.eq(Math.max(0, initialPoolSize - 1));
+      });
+    }
+  });
+
+  if (getTestDialect() !== 'sqlite' && getTestDialect() !== 'db2') {
+    it('works for long running transactions', async () => {
+      const sequelize2 = await prepareTransactionTest(sequelize);
+
+      const User = sequelize2.define('User', {
+        name: DataTypes.STRING,
+      }, { timestamps: false });
+
+      await sequelize2.sync({ force: true });
+      const t = await sequelize2.transaction();
+      let query = 'select sleep(2);';
+
+      switch (getTestDialect()) {
+        case 'postgres':
+          query = 'select pg_sleep(2);';
+          break;
+        case 'sqlite':
+          query = 'select sqlite3_sleep(2000);';
+          break;
+        case 'mssql':
+          query = 'WAITFOR DELAY \'00:00:02\';';
+          break;
+        default:
```

Comment:
Why not use 'select sleep(2);' here?

Reply:
It was already here in the .js version but looking through the docs I think mssql does not support `sleep()`

Reply:
I get that, but why not do `let query;` and add the default query under default? That way it's more clear what the actual query is instead of it being overwritten

Reply:
But this is not a blocking issue

Reply:
Aaah you pointed at `default`. I agree, I'll change it

Reply:
Done